### PR TITLE
#184693449 verify user via email

### PR DESCRIPTION
**Pull request by nkurunziza1**
#### What does this PR do?

- [x] A User  should be verified via email:
  - After signup the user receive an verification email
  - After verification the user allowed to login 
  - if user login with google  there is no need to verify

#### Description of Task to be completed?

- [x] A User  should be verified via email:

   -A user should be verified  via email
 
#### How should this be manually tested?

- Clone the repository 
- Change the directory to destructors-ec-bn by running `cd destructors-ec-bn` in you terminal
- Checkout ft-user-email-verify-184693449 branch by running `git checkout ft-user-email-verify-184693449` in you terminal
- open .env.example to check how you can create you .env file 
- Create your .env file
- Then install dependencies and devdependencies used by running `npm install` in you terminal
- To create the models used in this project run `npm migrate`
- Run the server by running `npm run dev` in your terminal
- open you browser and in URL field type http://localhost:PORT/api-docs (the PORT is the port you have written in your .env file. The default port is 3000)
- swagger page will open the navigate under  the products tag 
- you can post all user using /api/v1/signup and go to the gmail user and then click to the link so as to verify

#### What are the relevant pivotal tracker/Trello stories?

#184637449

#### how token should be written

Screenshot (72)

#### Questions:

### CODE REVIEW

**Review by katros1 — APPROVED** [review]
try to raise your coverage

**Review by buka1calvin — APPROVED** [review]

**Comment by mjeanbosco19** [comment]
I see some checks are failing, Could you please try to fix it?

**Review by JENNY-257 — APPROVED** [review]

**Review by mjeanbosco19 — APPROVED** [review]

File: src/controllers/user.controller.js [hunk 1]
```diff
@@ -50,7 +75,10 @@ const loginUser = async (req, res, next) => {
       if (!foundUser) {
         return res.status(404).json({ message: 'User not found' });
       }
-
+      if(foundUser.isEmailVerified == false){
+        console.log({message: foundUser.esEmailVerified})
```
Comment (Abdulkeza) [inline src/controllers/user.controller.js:79]: remove this console

File: src/controllers/user.controller.js [hunk 2]
```diff
@@ -61,19 +89,18 @@ const loginUser = async (req, res, next) => {
...
       const otp=generateOTP()
-       if(foundUser.role=="seller"){
+       if(foundUser.role=='seller'){
         OTP.otp=otp;
    await OTP.create({
          otp:otp,
          email:foundUser.email
        })
-       console.log("connected successfully!")
+       console.log('connected successfully!')
```
Comment (Abdulkeza) [inline src/controllers/user.controller.js:103]: remove this as well

File: src/controllers/user.controller.js [hunk 3]
```diff
@@ -106,7 +135,7 @@ const loginUser = async (req, res, next) => {
    const userEml = req.body.email;
    const user = await findUserByEmail(userEml);
     if (user == false) {
-      res.status(404).json("User not found");
+      res.status(404).json('User not found');
```
Comment (Abdulkeza) [inline src/controllers/user.controller.js:138]: please send a good-formated response

ex:  res.status(404).json({message: 'User not found'})

File: src/controllers/user.controller.js [hunk 4]
```diff
@@ -133,7 +162,7 @@ const loginUser = async (req, res, next) => {
     const token = req.params.token;
       const payload = verfyToken(token, process.env.JWT_SECRET)
 
-      console.log("payload", payload)
+      console.log('payload', payload)
```
Comment (Abdulkeza) [inline src/controllers/user.controller.js:165]: remove this as well

File: src/controllers/user.controller.js [hunk 5]
```diff
@@ -163,7 +192,7 @@ const editUserProfile = async(req,res)=>{
     const userEmail = req.user.email
     const decodeUser =await findUserByEmail(userEmail);
 
-    if(!decodeUser) return res.status(401).json("user not found");
+    if(!decodeUser) return res.status(401).json('user not found');
```
Comment (Abdulkeza) [inline src/controllers/user.controller.js:195]: 1. format this response as well
2. 401 status code if for Unauthorized NOT not found

**Pull request merged**